doc/requirements-doc.txt

```diff
@@ -22,7 +22,7 @@ sphinx-remove-toctrees==0.0.3
 autodoc_pydantic==1.6.1
 sphinx_design==0.4.1
 
-pydantic<2 # Pydantic is required by autodoc_pydantic, but must be <2 for ray
+pydantic==2.4.2
```

Comment:
you might need to follow the instructions (Updating dependencies and recompiling) in https://www.notion.so/anyscale-hq/OSS-Python-dependency-management-f32633b0018c423f927727807ea9da08

Maybe even just unpin the version and let the compiler find a suitable version

Reply:
I'm attempting to follow those 😓

Reply:
> Maybe even just unpin the version and let the compiler find a suitable version

it's unpinned in setup.py

Reply:
ah yeah he was referring to in our internal build system for the pinned test requirements